### PR TITLE
CAMEL-19298 Snmp: version 3 is not supported for several actions 

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://issues.apache.org/jira/browse/CAMEL-19298

PR contains basic tests for the combinations of actions and versions.

# Description

# Target

- [ ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [ ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [ ] I checked that each commit in the pull request has a meaningful subject line and body.

- [ ] I formatted the code using `mvn -Pformat,fastinstall install && mvn -Psourcecheck`

### CODE REVIEW

File: components/camel-snmp/src/test/java/org/apache/camel/component/snmp/TrapTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        LOG.info("Sending pdu " + trap);
+        Endpoint endpoint = context.getEndpoint("direct:snmptrapV" + version);
+        Exchange exchange = endpoint.createExchange();
+        exchange.getIn().setBody(trap);
+        Producer producer = endpoint.createProducer();
+        producer.process(exchange);
+
+        synchronized (this) {
+            Thread.sleep(1000);
+        }
```
Comment (orpiske) [inline components/camel-snmp/src/test/java/org/apache/camel/component/snmp/TrapTest.java:61]: Can we do this without the `Thread.sleep`, please? We already have plenty of those making out tests artificially slow.

**Review by orpiske — CHANGES_REQUESTED** [review]

File: components/camel-snmp/src/test/java/org/apache/camel/component/snmp/TrapTest.java (hunk 1 above)
Comment (orpiske) [reply components/camel-snmp/src/test/java/org/apache/camel/component/snmp/TrapTest.java:61]: IOW: I think you can use Awaitility here.

Comment (JiriOndrusek) [reply components/camel-snmp/src/test/java/org/apache/camel/component/snmp/TrapTest.java:61]: I'll create another PR with change of this code. TBH I don't remember writing this, and definitely Awaitility should be better here,

Comment (orpiske) [reply components/camel-snmp/src/test/java/org/apache/camel/component/snmp/TrapTest.java:61]: Thanks @JiriOndrusek, much appreciated!

Comment (JiriOndrusek) [reply components/camel-snmp/src/test/java/org/apache/camel/component/snmp/TrapTest.java:61]: PR: https://github.com/apache/camel/pull/10102

**Comment by JiriOndrusek** [comment]
> @JiriOndrusek if this needs backports, then you are welcome to send PRs for those branches

I'm currently trying to find out whether the backport is required.  I think that it would help camel-quarkus. Once I confirm that, I'll create backport PRs. Thankns!

**Comment by JiriOndrusek** [comment]
Backport to 3.183x: https://github.com/apache/camel/pull/10106

**Pull request merged**